src/pwm_input.rs

```diff
@@ -95,38 +84,65 @@ where
     RawValues { arr: u16, presc: u16 },
 }
 
+#[cfg(any(
+    feature = "stm32f100",
+    feature = "stm32f103",
+    feature = "stm32f105",
+))]
+impl Timer<TIM1> {
+    pub fn pwm_input<REMAP, PINS, T>(
```

Comment:
I am not familiar with using pwm_input and qei. So the question based on comparison with ```pwm``` implementation. I assume it is not ```pwm_input<REMAP, P, PINS, T>``` just because only (Ch1, Ch2) are allowed for pwm_input mode as well as QEI mode. Is this correct ? Or more configurations are supported, but this is current limitation of stm32f1xx_hal ?